File: component-annotation-compiler/src/main/java/q/rorbin/component/model/ComponentServiceType.java
package q.rorbin.component.model;

import java.util.List;
import java.util.Objects;

/**
 * @author 邱长海
 */
public class ComponentServiceType {
    private Class name;
    private String version;
    private List<Class> interfaces;

    public ComponentServiceType(Class name, String version, List<Class> interfaces) {
        this.name = name;
        this.version = version;
        this.interfaces = interfaces;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public List<Class> getInterfaces() {
        return interfaces;
    }

    public void setInterfaces(List<Class> interfaces) {
        this.interfaces = interfaces;
    }

    public Class getName() {
        return name;
    }

    public void setName(Class name) {
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ComponentServiceType that = (ComponentServiceType) o;
        return name.getName().equals(that.name.getName());
    }

    @Override
    public int hashCode() {
        return Objects.hash(name.getName());
    }
}

File: component-annotation-compiler/src/main/java/q/rorbin/component/model/ComponentServiceType.kt
package q.rorbin.component.model

import java.util.*

/**
 * @author changhai.qiu
 */
class ComponentServiceType(val name: Class<Any>, val version: String, val interfaces: List<Class<Any>>) {
    override fun hashCode(): Int {
        return Objects.hash(name.name)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || this::class.java != other::class.java) return false
        val that = other as ComponentServiceType
        return name.name == that.name.name
    }
}